src/main/java/com/box/sdk/BoxFile.java

```diff
@@ -649,7 +662,7 @@ public boolean canUploadVersion(String name) {
      */
     public boolean canUploadVersion(String name, long fileSize) {
 
-        URL url = CONTENT_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        URL url = getDownloadUrl();
```

Comment:
How about `uploadNewVersion` shouldn't it be changed there too?

Reply:
Upload new version is not checking anything before initialising upload. Why we would need to change it?

Reply:
So if we get upload URL by `URL url = getDownloadUrl(); ` so shouldn't upload itself url be same as checked?